File: data-structures/arrays/vb/array_1D_search.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program


Module Program

    ' The Main subroutine is the entry point for all VB programs
    Sub Main()
        ' Declare and initialise the array of words
        Dim spellingWords = New String(4) {"me", "do", "it", "her", "him"}

        ' Ask the user for a word to find
        Console.WriteLine("Enter a word to find:")
        Dim wordToFind As String = Console.ReadLine()

        ' Check if the word matches a value in the spelling array
        Dim found = CheckWordExists(spellingWords, wordToFind)

        ' Output the search results
        If found Then
            Console.WriteLine("Word found!")
        Else
            Console.WriteLine("Word not found in the array")
        End If
    End Sub


    ' Return Boolean True if search word is found in the array
    Function CheckWordExists(ByVal spellingWords As String(), ByVal searchWord As String) As Boolean
        Dim i = 0
        Dim isFound = False

        ' Repeat while the search word has not been found
        While Not isFound AndAlso i < spellingWords.Length
            ' Check if the words match
            If spellingWords(i).Equals(searchWord) Then
                isFound = True
            End If
            i += 1 ' Increment the counter
        End While

        Return isFound
    End Function


End Module

File: data-structures/arrays/java/array_1D_search.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class Arrays1D  {
    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        // Declare and initialise the array of words
        String[] spellingWords = {
            "me", "do", "it", "her", "him"
        };

        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        // Ask the user for a word to find
        System.out.println("Enter a word to find:");
        String wordToFind = scanner.nextLine();

        // Check if the word matches a value in the spelling array
        boolean found = checkWordExists(spellingWords, wordToFind);

        // Output the search results
        if (found) {
            System.out.println("Word found!");
        }
        else {
            System.out.println("Word not found in the array");
        }
    }

    // Return Boolean true if search word is found in the array
    public static boolean checkWordExists(String[] spellingWords, String searchWord) {
        int i = 0;
        boolean isFound = false;

        // Repeat while the search word has not been found
        while (!isFound && i < spellingWords.length) {
            // Check if the words match
            if (spellingWords[i].equals(searchWord)) {
                isFound = true;
            }
            i += 1; // Increment the counter
        }

        return isFound;
    }
}
